### ISSUE TITLE
Further Development Ideas

### ISSUE THREAD

**Issue by JulianKahnert** [jffry/rocket-r60v#1]
I just stumbled across this repository in an attempt to reverse-engineer the R60V myself. Kudos for making your findings available!

Do you have any plans on further extending the functionality of the code, like a CLI or a webinterface? How are you currently using it? Only via the android emulator / modified APK?

Thanks again for publishing this!

**Comment by jffry** [comment]
Hi Julian! The R60V I've been working with actually belongs to a friend who lives a couple thousand miles away from me, so I've been tinkering on it whenever I visit.

My friend doesn't like that the the auto-on, auto-standby feature forces you to use the same time on weekdays and weekends, so our humble goal is ultimately to have something on a Raspberry Pi that we can use to interact with the machine.  My goal for this repo is to keep it to tools & code for interfacing at a high level (e.g. "go to standby" or "adjust boiler temp to XX ºF"), while the Pi-based tooling will likely be a separate interface.

I _think_ I understand how the (simple) protocol works now - it's basically just reading or writing blocks of memory based on a 16-bit offset and 16-bit length.  My goal this weekend is to write up everything I've discovered (like what you can read/write and where different settings live).

I did add a few tools for connecting to the machine and sending commands - see `src/server/interactive.ts`, though I'd advise being super careful about sending any _write_ commands.  But, I'm happy to report that I am capable of sending a command (`w004A000101` + checksum) that puts the machine into standby, so stay tuned!

**Comment by JulianKahnert** [comment]
Hello Jeffrey, thank you for your answer! I am following your commits very closely and i'm impressed of the reverse engineered protocol. At the moment, i'm running in problems with the connection between my PI and the machine. Do you have some advice of your connection procedure? E.g. are you starting the machine or the PI first? Is there some way to reset the network interface of the machine?
I figured out, that my machine just talks on the first WLAN channel, which seemed to be problematic with some of my devices. Furthermore, the DHCP server on the R60V does not work properly. Have you experienced the same kind of problems?
Best regards, Julian

**Comment by JulianKahnert** [comment]
Hello Jeffrey, I figured out that Rocket uses [this WLAN chip](http://ww1.microchip.com/downloads/en/DeviceDoc/rn-131-ds-v3.2r.pdf) internally. It has an external antenna port, but it is not used. So I came up with the idea to put the WLAN Stick (which is conntected to the RaspberryPI) inside the R60V. This makes the hole setup much more reliable, maybe it helps you with your further development too.
Now I'm finally able to read and write the hole [state of the machine](https://github.com/JulianKahnert/RocketAPI/).
Best regards, Julian
